dom.bs

```diff
@@ -2001,54 +2001,74 @@ can be used to explore this matter in more detail.
+<p><dfn export id=concept-node>Nodes</dfn> are objects that <a>implement</a> {{Node}}.
+<a for=/>Nodes</a> <a>participate</a> in a <a>tree</a>, which is known as the
 <dfn export id=concept-node-tree>node tree</dfn>.
 
-<p>A <a>node tree</a> is constrained as follows, expressed as a relationship between the type of
-<a>node</a> and its allowed <a>children</a>:
+<div class=note>
+ <p>In practice you deal with more specific objects.
+
+ <p>Objects that <a>implement</a> {{Node}} also implement an inherited interface: {{Document}},
+ {{DocumentType}}, {{DocumentFragment}}, {{Element}}, {{CharacterData}}, or {{Attr}}.
+
+ <p>Objects that implement {{DocumentFragment}} sometimes implement {{ShadowRoot}}.
+
+ <p>Objects that <a>implement</a> {{Element}} also typically implement an inherited interface, such
+ as {{HTMLAnchorElement}}.
+
+ <p>Objects that implement {{CharacterData}} also implement an inherited interface: {{Text}},
+ {{ProcessingInstruction}}, or {{Comment}}.
+
+ <p>Objects that implement {{Text}} sometimes implement {{CDATASection}}.
+</div>
+
+<p>For brevity, this specification refers to an object that <a>implements</a> {{Node}} and an
+inherited interface <code><var>NodeInterface</var></code>, as a
+<code><var>NodeInterface</var></code> <a for=/>node</a>.
+
+<p>A <a>node tree</a> is constrained as follows, expressed as a relationship between a
+<a for=/>node</a> and its allowed <a for=tree>children</a>:
 
 <dl>
  <dt>{{Document}}
  <dd>
   <p>In <a>tree order</a>:
   <ol>
-   <li><p>Zero or more nodes each of which is {{ProcessingInstruction}} or {{Comment}}.
-   <li><p>Optionally one {{DocumentType}} node.
-   <li><p>Zero or more nodes each of which is {{ProcessingInstruction}} or {{Comment}}.
-   <li><p>Optionally one {{Element}} node.
-   <li><p>Zero or more nodes each of which is {{ProcessingInstruction}} or {{Comment}}.
+   <li><p>Zero or more {{ProcessingInstruction}} or {{Comment}} <a for=/>nodes</a>.
+   <li><p>Optionally one {{DocumentType}} <a for=/>node</a>.
+   <li><p>Zero or more {{ProcessingInstruction}} or {{Comment}} <a for=/>nodes</a>.
+   <li><p>Optionally one {{Element}} <a for=/>node</a>.
+   <li><p>Zero or more {{ProcessingInstruction}} or {{Comment}} <a for=/>nodes</a>.
   </ol>
+
  <dt>{{DocumentFragment}}
  <dt>{{Element}}
- <dd><p>Zero or more nodes each of which is {{Element}}, {{Text}}, {{ProcessingInstruction}}, or
- {{Comment}}.
+ <dd><p>Zero or more {{Element}} or {{CharacterData}} <a for=/>nodes</a>.
+
  <dt>{{DocumentType}}
- <dt>{{Text}}
- <dt>{{ProcessingInstruction}}
- <dt>{{Comment}}
+ <dt>{{CharacterData}}
+ <dt>{{Attr}}
  <dd><p>None.
 </dl>
 
-<p>To determine the <dfn export id=concept-node-length for=Node>length</dfn> of a <a>node</a>
-<var>node</var>, switch on <var>node</var>:
+<p class=note>{{Attr}} <a for=/>nodes</a> <a>participate</a> in a <a>tree</a> for historical
+reasons; they never have a <a for=tree>parent</a> or <a for=tree>children</a> and are therefore
```

Comment:
I think "their parent is always null" is a bit more technically accurate. Because https://whatpr.org/dom/1004.html#concept-tree-participate says "An object that participates in a tree has a parent", which seems to contradict this.

I guess technically "never have children" should be "always have an empty set for their children", but that one bothers me much less for some reason.